eviltrout/ember-performance
# Add Ember 4.0 support, use prod builds, more benchmarks, UX improvements

See individual commits for more details. Intended to be 'rebased-and-merged', not squashed.

@eviltrout if you'd prefer that I split this into multiple PRs then that's fine, just let me know.

---
- **Remove 'latest' version functionality**
   http://builds.emberjs.com/ hasn't been updated for many years - this feature is broken

-  **UX: Sort benchmarks alphabetically**

-  **Use prod builds, add v3.28.8 and v4.0.1, add build instructions**
    
    Increasingly, Ember relies on the ember-cli defaults, and other packages. The new instructions in the README will ensure we get all the default packages/defaults included in the bundle, and should make it easier to add new versions in the future.
    
    The new instructions also give us a true 'production' build. The existing builds we're using have been taken directly from the `ember-source` package, and include debug-only logic which may affect performance.
    
    This commit replaces all existing versions with ember-cli production builds, as well as introducing 3.8.3 and 4.0.1

- **Update templates and remove use of globals**
    
    These syntaxes and globals are deprecated in Ember 3, and removed in Ember 4. Templates are also updated to use the 'named argument syntax' (`@arg`) so that they're compatible with both 'classic' Ember components and Glimmer components.

 - **Remove 3.4.8**
    
    This old version of Ember doesn't seem to work with native Glimmer components. Keeping support would require adding special-cases for this specific version.

- **Add benchmarks for Glimmer components**
    
    These are the new defaults in the latest versions of Ember, so it's useful to track their performance across versions, and in comparison to Classic components.

-  **Include error-range in charts**

-  **Add complex-html benchmark for raw handlebars**
